src/crosswalk/model.py

```diff
@@ -529,44 +529,37 @@ def get_cov_names(self) -> List[str]:
                 cov_names.extend([f'{model.cov_name}_spline_{i}' for i in range(model.num_vars)])
         return cov_names
 
-    def create_result_df(self) -> pd.DataFrame:
-        """Create result data frame.
+    def create_fixed_df(self) -> pd.DataFrame:
+        """Create fixed effects result data frame.
 
         Returns:
-            pd.DataFrame: Data frame that contains the result.
+            pd.DataFrame: Fixed effects result data frame.
         """
         # column of dorms
         dorms = np.repeat(self.cwdata.unique_dorms, self.num_vars_per_dorm)
         cov_names = self.get_cov_names()
         cov_names *= self.cwdata.num_dorms
 
         # create data frame
-        df = pd.DataFrame({
+        fixed_df = pd.DataFrame({
             'dorms': dorms,
             'cov_names': cov_names,
             'beta': self.beta,
             'beta_sd': self.beta_sd,
         })
-        if self.use_random_intercept:
-            gamma = np.hstack((self.lt.gamma, np.full(self.num_vars - 1, np.nan)))
-            re = np.hstack((self.lt.u, np.full((self.cwdata.num_studies, self.num_vars - 1), np.nan)))
-            df['gamma'] = gamma
-            for i, study_id in enumerate(self.cwdata.unique_study_id):
-                df[study_id] = re[i]
-
-        return df
+        return fixed_df
 
-    def save_result_df(self, folder: str, filename: str = 'result.csv'):
-        """Save result.
+    def create_random_df(self) -> pd.DataFrame:
+        """Create random effects result data frame.
 
-        Args:
-            folder (str): Path to the result folder.
-            filename (str): Name of the result. Default to `'result.csv'`.
+        Returns:
+            pd.DataFrame: Random effects result data frame.
         """
-        if not filename.endswith('.csv'):
-            filename += '.csv'
-        df = self.create_result_df()
-        df.to_csv(folder + '/' + filename, index=False)
+        random_df = pd.DataFrame({
```

Comment:
@kels271828 Here should we differentiate behavior based `self.use_random_intercept` is true or false?

Reply:
Yes, I was wondering about that! Do we want to throw an exception, create a warning and return None, or just return None? Or return some other value?

Reply:
From a user perspective it is nice to be reminded about the model setup. So we can print out a warning says that there isn't random intercept in the model and return the data frame. I think the current code will still work, but all the gamma or the random effects will be zero.

Reply:
Added a warning!